external/kafka-0-10/pom.xml

```diff
@@ -49,8 +49,36 @@
     </dependency>
     <dependency>
       <groupId>org.apache.kafka</groupId>
+      <artifactId>kafka-clients</artifactId>
```

Comment:
Hm, ideally we unify this with the similar declaration already in the Spark SQL Kafka 0.10 POM but they don't have a parent that is Kafka-specific. At least they should be consistent. Why are the excludes needed? We also probably want a refactored `kafka.version` property in each to avoid repeating the version.

Finally can you explain the change a bit and make a JIRA? http://spark.apache.org/contributing.html